src/DynamoCoreWpf/Views/Core/GeometryScalingPopup.xaml.cs

```diff
@@ -0,0 +1,111 @@
+using System.Windows.Controls.Primitives;
+using Dynamo.ViewModels;
+using ViewModels.Core;
+using System.Linq;
+using Dynamo.Logging;
+using System;
+using Res = Dynamo.Wpf.Properties.Resources;
+using System.Windows.Input;
+
+namespace Dynamo.Views
+{
+    public partial class GeometryScalingPopup : Popup
+    {
+        private GeometryScalingViewModel viewModel;
+        private DynamoViewModel dynamoViewModel;
+        
+        public GeometryScalingPopup(DynamoViewModel dynViewModel)
+        {
+            InitializeComponent();
+
+            if (viewModel == null && dynViewModel.GeoScalingViewModel != null)
+                viewModel = dynViewModel.GeoScalingViewModel;
+            DataContext = viewModel;
+            dynamoViewModel = dynViewModel;
+        }
+
+        /// <summary>
+        /// This event is generated every time the user clicks a Radio Button in the Geometry Scaling section
+        /// The method just get the Radio Button clicked and saves the ScaleValue selected
+        /// This are the values used for the scales:
+        /// - 2 - Small
+        ///   0 - Medium (Default)
+        ///   2 - Large
+        ///   4 - Extra Large
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Geometry_Scaling_Checked(object sender, System.Windows.RoutedEventArgs e)
+        {
+            var selectedButton = sender as Button;
+            if (selectedButton == null) return;
+            //var buttons = BaseGrid.Children.OfType<Button>();
+            var buttons = GeometryScalingRadiosPanel.Children.OfType<Button>();
+
+            int index = 0;
+
+            ////We need to loop all the radiobuttons in the GeometryScaling section in order to find the index of the selected one
+            foreach (var button in buttons)
+            {
+                if (button == selectedButton)
+                {
+                    viewModel.ScaleValue = GeometryScalingOptions.ConvertUIToScaleFactor(index);
+                    break;
+                }
+                index++;
+            }          
+            RunGraphWhenScaleFactorUpdated();
+            this.IsOpen = false;
+        }
+
+        /// <summary>
+        /// This method will run the graph only if the Geometry Scaling was updated otherwise will not be executed
+        /// </summary>
+        private void RunGraphWhenScaleFactorUpdated()
+        {
+            //If the new radio button selected (ScaleValue) is different than the current one in Dynamo, we update the current one
+            if (dynamoViewModel.ScaleFactorLog != viewModel.ScaleValue)
+            {
+                dynamoViewModel.ScaleFactorLog = (int)viewModel.ScaleValue;
+                dynamoViewModel.CurrentSpace.HasUnsavedChanges = true;
+
+                //Due that binding are done before the contructor of this class we need to execute the Log only if the viewModel was assigned previously
```

Comment:
`constructor`

Reply:
Fixed comment in the next commit: https://github.com/DynamoDS/Dynamo/pull/13656/commits/789f51c4a3c106156940924585e1ca40abd9842b